src/wrappers/themis/python/pythemis/smessage.py

```diff
@@ -24,82 +25,115 @@
-        res = themis.themis_secure_message_unwrap(
+        res = themis.themis_secure_message_decrypt(
             self.private_key, len(self.private_key),
             self.peer_public_key, len(self.peer_public_key),
             message, len(message), plain_message, byref(plain_message_length))
         if res != THEMIS_CODES.SUCCESS:
-            raise ThemisError(res, "Secure Message failed decrypting")
+            raise ThemisError(res, "Secure Message failed to decrypt")
         return string_at(plain_message, plain_message_length.value)
 
 
 def ssign(private_key, message):
     encrypted_message_length = c_int(0)
-    res = themis.themis_secure_message_wrap(
-        private_key, len(private_key), None, 0, message, len(message),
+    res = themis.themis_secure_message_sign(
+        private_key, len(private_key), message, len(message),
         None, byref(encrypted_message_length))
     if res != THEMIS_CODES.BUFFER_TOO_SMALL:
-        raise ThemisError(res, "Secure Message failed singing")
+        raise ThemisError(res, "Secure Message failed to sign")
     encrypted_message = create_string_buffer(encrypted_message_length.value)
-    res = themis.themis_secure_message_wrap(
-        private_key, len(private_key), None, 0, message, len(message),
+    res = themis.themis_secure_message_sign(
+        private_key, len(private_key), message, len(message),
         encrypted_message, byref(encrypted_message_length))
     if res != THEMIS_CODES.SUCCESS:
-        raise ThemisError(res, "Secure Message failed singing")
+        raise ThemisError(res, "Secure Message failed to sign")
     return string_at(encrypted_message, encrypted_message_length.value)
 
 
 def sverify(public_key, message):
     plain_message_length = c_int(0)
-    res = themis.themis_secure_message_unwrap(
-        None, 0, public_key, len(public_key), message, len(message),
+    res = themis.themis_secure_message_verify(
+        public_key, len(public_key), message, len(message),
         None, byref(plain_message_length))
     if res != THEMIS_CODES.BUFFER_TOO_SMALL:
-        raise ThemisError(res, "Secure Message failed verifying")
+        raise ThemisError(res, "Secure Message failed to verify")
     plain_message = create_string_buffer(plain_message_length.value)
-    res = themis.themis_secure_message_unwrap(
-        None, 0, public_key, len(public_key), message, len(message),
+    res = themis.themis_secure_message_verify(
+        public_key, len(public_key), message, len(message),
         plain_message, byref(plain_message_length))
     if res != THEMIS_CODES.SUCCESS:
-        raise ThemisError(res, "Secure Message failed verifying")
+        raise ThemisError(res, "Secure Message failed to verify")
 
     return string_at(plain_message, plain_message_length.value)
 
 
+class THEMIS_KEY(IntEnum):
+    INVALID = 0
+    RSA_PRIVATE = 1
+    RSA_PUBLIC = 2
+    EC_PRIVATE = 3
+    EC_PUBLIC = 4
+
+
+def _public_key(key):
```

Comment:
👍

Reply:
better to make this function public to allow use it for apps

Reply:
and better to rename to something like `validate_public_key` because `_public_key` looks like getter function

Reply:
Mmm... I'm not against it but I think there's a better way.

These two functions are effectively internal helpers which implement a particular check specifically for Secure Message needs. However, the user may have different needs that that. I think that for general use it would be better to export C functions `themis_is_valid_asym_key` and `themis_get_asym_key_kind` which provide a bit more information about the keys. Then the user could implement their of `is_valid_public_key` function the way they want.

Reply:
will be great. but not only `is_valid_public_key`, I think `is_valid_private_key` will be useful for users too.

Reply:
I think it's a good idea overall, but let's do it later if you don't mind: #409

Reply:
sure